### PR TITLE
Fix attachment regex when no markdown text

### PR DESCRIPTION
**Pull request by alanhdu**
Right now, the regex is suited for things like:
```markdown
This is an image ![logo-64x64.png](attachment:cbe9739f-7fcd-4ed5-866f-80d693d30fe8.png)
```
which gets translated into:
```rst
This is an image: |logo-64x64.png|

.. |logo-64x64.png| image:: attachment:cbe9739f-7fcd-4ed5-866f-80d693d30fe8.png
```

However, if you have an image without an text in front of it like:
```markdown
![logo-64x64.png](attachment:cbe9739f-7fcd-4ed5-866f-80d693d30fe8.png)
```

Then the rst looks like:
```rst
.. image:: attachment:cbe9739f-7fcd-4ed5-866f-80d693d30fe8.png
   :alt: logo-64x64.png
```

Crucially, this means that there is no `|<filename.png>| ` to match, which causes the regex to fail.  This is a simple solution which just makes the alt-text matching optional in the regex.

Fixes https://github.com/spatialaudio/nbsphinx/issues/525

### CODE REVIEW

**Comment by alanhdu** [comment]
I'm not exactly sure the right way to test this -- I checked manually that this fixes rendering my specific notebook. I added an example of this behavior that to the `markdown-cells.ipynb` notebook -- is that the best way to test this change?

**Comment by mgeier** [comment]
Thanks for this PR!

> is that the best way to test this change?

The "best way" is quite a strong claim, I'm not sure about that, but I think it's a good enough way to test it.
 
>  I haven't been able to build the docs locally based on some `bibtex` configuration error.

I've just merged #524 which should fix (or rather postpone) the error, can you please rebase?

**Comment by alanhdu** [comment]
Rebased!

**Comment by mgeier** [comment]
Thanks, the rendered pages is available at https://nbsphinx--526.org.readthedocs.build/en/526/markdown-cells.html#Cell-Attachments.

File: doc/markdown-cells.ipynb [hunk 1]
```diff
@@ -332,7 +335,9 @@
     "\n",
     "    ![a stick figure](attachment:stickfigure.png)\n",
     "\n",
-    "This is a cell attachment: ![a stick figure](attachment:stickfigure.png)\n",
+    "These are cell attachments: ![a stick figure](attachment:stickfigure.png)\n",
+    "\n",
+    "![logo-64x64.png](attachment:98a753bb-02aa-42e8-81da-6a5c4f9b8eb5.png)\n",
```
Comment (mgeier) [inline doc/markdown-cells.ipynb:340]: Does this file name have some meaning?
What about using a more descriptive name, like e.g. `logo-64x64.png`?

The `alt` text (between the brackets) should be for accessibility, I don't think that `logo-64x64.png` really helps with that, right?

**Comment by alanhdu** [comment]
I believe the test failure is a flaky one:

```
(line 1573) broken    https://developer.mozilla.org/en-US/docs/Tools/Web_Console#Opening_the_Web_Console - Anchor 'Opening_the_Web_Console' not found
```

but when I click on the link, it seems to work fine.

**Comment by mgeier** [comment]
Thanks a lot!

Yes, the test failure is unrelated, I'm trying to fix it in #528.

**Pull request merged**